aws/resource_aws_cloudwatch_log_group.go

```diff
@@ -40,9 +41,10 @@ func resourceAwsCloudWatchLogGroup() *schema.Resource {
 			},
 
 			"retention_in_days": {
-				Type:     schema.TypeInt,
-				Optional: true,
-				Default:  0,
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntInSlice([]int{1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653}),
```

Comment:
Zero (never expire) could also be in use currently since the resource handles it properly with `d.GetOk()` 👍 

```suggestion
				ValidateFunc: validation.IntInSlice([]int{0, 1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653}),
```